### PR TITLE
[Impeller] fix remaining Validation errors.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/145039

Some of the errors were due to unconsumed fragment shader inputs. Enabling VK_KHR_maintenance4 relaxes this restriction.

### CODE REVIEW

File: impeller/scene/shaders/unlit.frag [hunk 1]
```diff
@@ -17,8 +17,13 @@ in vec4 v_color;
 
 out vec4 frag_color;
 
+float consume_unused() {
```
Comment (jonahwilliams) [inline impeller/scene/shaders/unlit.frag:20]: This makes the shader inputs technically consumed until we figure out how to handle this in general.

File: impeller/entity/entity_unittests.cc [hunk 2]
```diff
@@ -2251,11 +2254,9 @@ TEST_P(EntityTest, RuntimeEffect) {
     ASSERT_TRUE(runtime_stage->IsDirty());
     expect_dirty = true;
 
-    callback(*content_context, mock_pass);
+    // callback(*content_context, mock_pass);
```
Comment (gaaclarke) [inline impeller/entity/entity_unittests.cc:2257]: stray comment

**Review by matanlurey — APPROVED** [review]

File: impeller/runtime_stage/runtime_stage_unittests.cc [hunk 3]
```diff
@@ -277,7 +277,17 @@ TEST_P(RuntimeStageTest, CanCreatePipelineFromRuntimeStage) {
   auto vertex_descriptor = std::make_shared<VertexDescriptor>();
   vertex_descriptor->SetStageInputs(VS::kAllShaderStageInputs,
                                     VS::kInterleavedBufferLayout);
-  vertex_descriptor->RegisterDescriptorSetLayouts(VS::kDescriptorSetLayouts);
+
+  std::array<DescriptorSetLayout, 2> descriptor_set_layouts = {
+      VS::kDescriptorSetLayouts[0],
+      DescriptorSetLayout{
+          64,                              // binding = 0
```
Comment (gaaclarke) [inline impeller/runtime_stage/runtime_stage_unittests.cc:284]: c++ style: use checked name comments `/*foo=*/`

**Review by gaaclarke — COMMENTED** [review]
lgtm, just a couple of notes

File: impeller/entity/entity_unittests.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/entity/entity_unittests.cc]: Fixed

File: impeller/runtime_stage/runtime_stage_unittests.cc (hunk 3 above)
Comment (jonahwilliams) [reply impeller/runtime_stage/runtime_stage_unittests.cc]: Switched this to use initializers instread

**Pull request merged**